onnxruntime.sh

```diff
@@ -18,24 +26,138 @@ prepend_path:
-      -DPYTHON_EXECUTABLE=$(python3 -c "import sys; print(sys.executable)")           \
-      -Donnxruntime_BUILD_UNIT_TESTS=OFF                                              \
-      -Donnxruntime_PREFER_SYSTEM_LIB=ON                                              \
-      -Donnxruntime_BUILD_SHARED_LIB=ON                                               \
-      -DProtobuf_USE_STATIC_LIBS=ON                                                   \
-      ${PROTOBUF_ROOT:+-DProtobuf_LIBRARY=$PROTOBUF_ROOT/lib/libprotobuf.a}           \
-      ${PROTOBUF_ROOT:+-DProtobuf_LITE_LIBRARY=$PROTOBUF_ROOT/lib/libprotobuf-lite.a} \
-      ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_LIBRARY=$PROTOBUF_ROOT/lib/libprotoc.a}      \
-      ${PROTOBUF_ROOT:+-DProtobuf_INCLUDE_DIR=$PROTOBUF_ROOT/include}                 \
-      ${PROTOBUF_ROOT:+-DProtobuf_PROTOC_EXECUTABLE=$PROTOBUF_ROOT/bin/protoc}        \
-      ${RE2_ROOT:+-DRE2_INCLUDE_DIR=${RE2_ROOT}/include}                              \
-      ${BOOST_ROOT:+-DBOOST_INCLUDE_DIR=${BOOST_ROOT}/include}                        \
-      -DCMAKE_CXX_FLAGS="$CXXFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable -Wno-error=deprecated" \
-      -DCMAKE_C_FLAGS="$CFLAGS -Wno-unknown-warning -Wno-unknown-warning-option -Wno-error=unused-but-set-variable -Wno-error=deprecated"
+# Check ROCm build conditions
+if [[ -f /etc/redhat-release ]]; then
+  export ALMA_LINUX_MAJOR_VERSION=$(awk '{print $3}' /etc/redhat-release | cut -d. -f1)
+fi
+if [[ "$ALIBUILD_O2_FORCE_GPU" -eq 1 ]] || [[ "$ALIBUILD_ENABLE_HIP" -eq 1 ]] || \
+  ( ( [[ -z "$DISABLE_GPU" ]] || [[ "$DISABLE_GPU" -eq 0 ]] ) && \
+  ( command -v /opt/rocm/bin/rocminfo >/dev/null 2>&1 ) && \
+  [[ -d /opt/rocm/include/hiprand ]] && \
+  [[ -d /opt/rocm/include/hipblas ]] && \
+  [[ -d /opt/rocm/include/hipsparse ]] && \
+  [[ -d /opt/rocm/include/hipfft ]] && \
+  [[ -d /opt/rocm/include/rocblas ]] && \
+  [[ -d /opt/rocm/include/rocrand ]] && \
+  [[ -d /opt/rocm/include/miopen ]] && \
+  [[ -d /opt/rocm/include/rccl ]] && \
+  [[ -d /opt/rocm/lib/hipblaslt ]] && \
+  [[ -z "$ORT_ROCM_BUILD" ]] ) && \
+  ([[ -z "$ALMA_LINUX_MAJOR_VERSION" ]] || [[ "$ALMA_LINUX_MAJOR_VERSION" -eq 9 ]]); then
+  export ORT_ROCM_BUILD="1"
+  : ${ALIBUILD_O2_OVERRIDE_HIP_ARCHS:="gfx906,gfx908"}
+  export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/opt/rocm/lib
+else
+  export ORT_ROCM_BUILD="0"
+fi
+
+# Check CUDA build conditions
+if ( [[ "$ALIBUILD_O2_FORCE_GPU" -eq 1 ]] || [[ "$ALIBUILD_ENABLE_CUDA" -eq 1 ]] || \
+  ( ( [[ -z "$DISABLE_GPU" ]] || [[ "$DISABLE_GPU" -eq 0 ]] ) && \
+  ( command -v nvcc >/dev/null 2>&1 ) && \
+  [[ -f /usr/include/cudnn.h ]] && \
+  [[ -z "$ORT_CUDA_BUILD" ]] ) ) && \
+  [[ "$ORT_ROCM_BUILD" -eq 0 ]] && \
+  [[ -z "$ALMA_LINUX_MAJOR_VERSION" ]]; then
+  export ORT_CUDA_BUILD="1"
+  : ${ALIBUILD_O2_OVERRIDE_CUDA_ARCHS:="sm_86"}
+else
+  export ORT_CUDA_BUILD="OFF"
```

Comment:
If there are more commits coming, then maybe we can keep the naming consistent, ("ON" / "OFF" or 0/1)

Reply:
We can do that, but then we should probably follow the CMake convention where booleans are with ON / OFF, not 1 / 0.